promises/views.py

```diff
@@ -13,27 +13,24 @@
 promise_logger = logging.getLogger('promise')
 
 
-@api_view(['GET'])
-@permission_classes((AllowAny,))
+@api_view(['GET', 'POST'])
+@permission_classes((IsAuthenticatedOrReadOnly,))
 def promise_list(request):
-    promises = Promise.objects.all()
-    serializer = PromiseSerializerRead(promises, many=True)
-    return Response(serializer.data)
-
-
-@api_view(['POST'])
-@permission_classes((IsAuthenticated,))
-def promise_list(request):
-    promise_logger.info(request.user)
-    serializer = PromiseSerializerWrite(data=request.data)
-    if serializer.is_valid():
-        serializer.save()
-        return Response(serializer.data, status=status.HTTP_201_CREATED)
-    return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+    if request.method == 'GET':
+        promises = Promise.objects.all()
+        serializer = PromiseSerializerRead(promises, many=True)
+        return Response(serializer.data)
+    elif request.method == 'POST':
```

Comment:
IMO it is more readable to have
```
api_view(['POST'])
....
```
if there is an if loop inside the function call